File: jambeth/jambeth-merge/src/main/java/de/osthus/ambeth/merge/IEntityInstantiationExtensionExtendable.java
package de.osthus.ambeth.merge;

public interface IEntityInstantiationExtensionExtendable
{
	void registerEntityInstantiationExtension(IEntityInstantiationExtension entityInstantiationExtension, Class<?> type);

	void unregisterEntityInstantiationExtension(IEntityInstantiationExtension entityInstantiationExtension, Class<?> type);
}

File: ambeth/Ambeth.Merge/ambeth/merge/IEntityInstantiationExtensionExtendable.cs
using System;

namespace De.Osthus.Ambeth.Merge
{
    public interface IEntityInstantiationExtensionExtendable
    {
        void RegisterEntityInstantiationExtension(IEntityInstantiationExtension entityInstantiationExtension, Type type);

        void UnregisterEntityInstantiationExtension(IEntityInstantiationExtension entityInstantiationExtension, Type type);
    }
}